### PR TITLE
Improve implementation of get_available_triplets.

### PR DESCRIPTION
**Pull request by dapladoc**
`get_available_triplets` was rewritten in a loop-less manner by means of PyTorch and NumPy

### CODE REVIEW

**Review by AlekseySh — APPROVED** [review]
Looks good for me!

**Comment by AlekseySh** [comment]
@DaloroAT do you want to add anything?

File: oml/miners/inbatch_all_tri.py [hunk 1]
```diff
@@ -15,16 +17,19 @@ class AllTripletsMiner(ITripletsMinerInBatch):
 
     """
 
-    def __init__(self, max_output_triplets: int = maxsize):
+    def __init__(self, max_output_triplets: int = maxsize, device: str = "cuda"):
```
Comment (AlekseySh) [inline oml/miners/inbatch_all_tri.py:20]: set "cpu" as default please

**Review by AlekseySh — CHANGES_REQUESTED** [review]

File: oml/miners/inbatch_all_tri.py (hunk 1 above)
Comment (AlekseySh) [reply oml/miners/inbatch_all_tri.py:20]: it also may help to make tests green

Comment (dapladoc) [reply oml/miners/inbatch_all_tri.py]: Sorry, my bad

**Review by AlekseySh — APPROVED** [review]
LGFM

**Review by DaloroAT — APPROVED** [review]
Super

**Pull request merged**